### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Initial commit
* `json` output matches [aws_iam_policy_document](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) output
* Added complete/example

## why
* Allows compiling iam policies from a list of statements
* Allows merging iam policies from a source url

## references
* https://github.com/cloudposse/terraform-aws-helm-release/pull/1

### CODE REVIEW

File: README.yaml [hunk 1]
```diff
@@ -73,7 +73,7 @@ usage: |-
 
   ```hcl
   module "example" {
```
Comment (aknysh) [inline README.yaml:75]:
```suggestion
  module "iam_policy" {
```

File: README.yaml [hunk 2]
```diff
@@ -73,7 +73,7 @@ usage: |-
...
-    source = "https://github.com/cloudposse/terraform-example-module.git?ref=master"
+    source  = "git::https://github.com/cloudposse/terraform-aws-iam-policy.git?ref=master"
     example = "Hello world!"
```
Comment (aknysh) [inline README.yaml:77]: remove `example`, use the correct attributes for the module

Comment (aknysh) [reply README.yaml:77]: update

```
# Short description of this project
description: |-
  This is `terraform-example-module` project provides all the scaffolding for a typical well-built Cloud Posse module. It's a template repository you can
  use when creating new repositories.
```
and check README.yaml for other sections that need to be updated

File: outputs.tf [hunk 3]
```diff
@@ -1,14 +1,4 @@
...
-output "example" {
-  description = "Example output"
-  value       = module.this.enabled ? local.example : null
-}
-
-output "random" {
-  description = "Stable random number for this example"
-  value       = module.this.enabled ? join("", random_integer.example[*].result) : null
+output "json" {
+  description = "JSON body of the iam policy document"
```
Comment (aknysh) [inline outputs.tf:2]:
```suggestion
  description = "JSON body of the IAM policy document"
```

File: variables.tf [hunk 4]
```diff
@@ -1,4 +1,11 @@
-variable "example" {
-  description = "Example variable"
-  default     = "hello world"
+variable "iam_source_json_url" {
+  type        = string
+  description = "IAM source json policy to download"
```
Comment (aknysh) [inline variables.tf:3]:
```suggestion
  description = "IAM source JSON policy to download"
```

Comment (aknysh) [reply variables.tf:3]: explain (in the description and prob in README, why do we need the URL to download external policy JSON files.
Mention that it's used in `source_json`

File: variables.tf [hunk 5]
```diff
@@ -1,4 +1,11 @@
...
+  default     = null
+}
+
+variable "iam_policy_statements" {
+  type        = list(any)
+  description = "IAM policy"
```
Comment (aknysh) [inline variables.tf:9]:
```suggestion
  description = "List of IAM policy statements"
```

**Review by aknysh — DISMISSED** [review]
please see comments

**Review by jamengual — APPROVED** [review]

**Pull request merged**